### PR TITLE
AP_HAL_ChibiOS: Add on-board baro support to iFlight BeastH7 V2 target.

### PR DESCRIPTION
**Pull request by jimsynz**
iFlight's own website says that there is no barometer on this device,
however on my one there is a DPS310 connected on I2C.  This PR adds
support for the on-board baro but leaves it as optional.

### CODE REVIEW

**Comment by andyp1per** [comment]
hi @jimsynz - there is already a PR for this, can you try it - https://github.com/ArduPilot/ardupilot/pull/20461
There appear to be more changes than just baro

**Comment by jimsynz** [comment]
@andyp1per @20461 appears to be for the BeastF7 not the BeastH7

**Comment by andyp1per** [comment]
My apologies, is it the same IMU? It is not on the F7. There seem to be at least 3 variants of each now.

**Comment by jimsynz** [comment]
Yeah, their website says MPU6000 in the page title but BMI270 in the specs is right in my case.

**Comment by jimsynz** [comment]
@andyp1per I've just rebased this against the latest master.  Is there anything else I need to do to get this merged? It'd be nice if it went into the 4.2 release.

**Review by andyp1per — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/BeastH7v2/hwdef.dat [hunk 1]
```diff
@@ -20,8 +20,7 @@ APJ_BOARD_ID 1056
 PB4 TIM3_CH1 TIM3 PWM(2) GPIO(51) BIDIR   # 2
 PB5 TIM3_CH2 TIM3 PWM(3) GPIO(52)         # 3
 
-# only one I2C bus
-I2C_ORDER I2C2
+I2C_ORDER I2C1
```
[the remark this answers is not in the document]
Comment (jimsynz) [reply libraries/AP_HAL_ChibiOS/hwdef/BeastH7v2/hwdef.dat:23]: Updated. Thanks!

**Review by andyp1per — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Comment by andyp1per** [comment]
@rmackay9 probably a candidate for 4.2

**Pull request merged**